sdk/src/types/block/mana/allotment.rs

```diff
@@ -34,8 +31,8 @@ impl Ord for ManaAllotment {
 }
 
 impl ManaAllotment {
-    pub fn new(account_id: AccountId, mana: u64) -> Result<Self, Error> {
-        if mana > THEORETICAL_MANA_MAX {
+    pub fn new(account_id: AccountId, mana: u64, protocol_params: &ProtocolParameters) -> Result<Self, Error> {
+        if mana > protocol_params.mana_structure().max_mana() {
```

Comment:
Discussion for later as we have an issue for that but I'd call it mana_max for consistency